### PR TITLE
Arabic translation

### PR DESCRIPTION
**Pull request by jaksz**
Can someone render the Arabic `basics.md` file and see if the right-to-left text is displaying properly in Jekyll so far?

I added a `dir: rtl` property into the Arabic `basics.md` file, hopefully it helps somehow.

### CODE REVIEW

File: ar/lessons/basics/basics.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+البدء، أنواع البيانات الأساسية والعمليات الأساسية.
+
+{% include toc.html %}
+
+## البداء
+
+### تثبيت إليكسير
+
+يمكن إيجاد تعليمات التثبيت لكل نظام التشغيل على موقع elixir-lang.org في دليل [تثبيت إليكسير](http://elixir-lang.org/install.html).
```
Comment (jaksz) [inline ar/lessons/basics/basics.md]: I'm afraid it's gonna look this way in the rendered version as well.

**Comment by doomspork** [comment]
Hi @jaksz, this is what it currently renders as:

image

**Comment by jaksz** [comment]
Thanks for the screenshot.

1) It's not entirely correct as it should be aligned right, but this part so far is readable. Maybe aligning it right is a sinple setting away?

2) Could you screenshot the part where there is English text mixed with Arabic? This is where I assume might be the problems.

**Comment by jaksz** [comment]
To clarify 2), the problematic text would be in the basics file, the screenshot only shows the ToC.
Thanks!

**Comment by doomspork** [comment]
For 1 I _think_ we're going to need some kind of CSS flag, at least that's my take away from reading up on the topic.

@jaksz is this sufficient? 
image

**Comment by jaksz** [comment]
Yes, thanks for the screenshot. Unfortunately there is a big problem with the way this sentence is rendered, exactly the issue I described in the first post. Maybe it's a browser rendering issue? What browser are you using? Can you upload a HTML file?

**Comment by doomspork** [comment]
@jaksz I am using Chrome.

Here is the HTML as-is:

```html

<!DOCTYPE html>
<html lang="en-us">

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <link href="http://gmpg.org/xfn/11" rel="profile">

  <meta name="description" content="Lessons about the Elixir programming language, inspired by Twitter’s Scala School.">
  <meta name="author" content="Sean Callan">

  <meta property="og:url" content="https://elixirschool.com/">
  <meta property="og:site_name" content="ElixirSchool">
  <meta property="og:title" content="
  الأساسيات &middot; Elixir School

">
  <meta property="og:description" content="Lessons about the Elixir programming language, inspired by Twitter’s Scala School.">
  <meta property="og:image" content="https://elixirschool.com/assets/fb_share.jpg">

  <title>
    
  الأساسيات &middot; Elixir School


  </title>

  <!-- CSS -->
  <link type="text/css" rel="stylesheet" href="/assets/app.css">

  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">

  <script type="text/javascript">
  !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t){var e=document.createElement("script");e.type="text/javascript";e.async=!0;e.src=("https:"===document.location.protocol?"https://":"http://")+"cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(e,n)};analytics.SNIPPET_VERSION="3.1.0";
  analytics.load("C0p07Sw9j7mZnycKDCgPOAyqJR5hhyoQ");
  analytics.page()
  }}();
  </script>

</head>


  <body class="theme-custom-purple">
    <a href="https://github.com/doomspork/elixir_school"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://camo.githubusercontent.com/652c5b9acfaddf3a9c326fa6bde407b87f7be0f4/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f6769746875622f726962626f6e732f666f726b6d655f72696768745f6f72616e67655f6666373630302e706e67" alt="Fork me on GitHub" data-canonical-src="https://s3.amazonaws.com/github/ribbons/forkme_right_orange_ff7600.png"></a>

    
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      <h1>
        <a href="/">
          Elixir School
        </a>
      </h1>
      <p class="lead">دروس عن لغة برمجة إليكسير</p>
    </div>

    <nav class="sidebar-nav">
      
      
        
        
        <div class="sidebar-nav-group basics">
          <div class="sidebar-nav-header">
            <i class="fa  fa-chevron-down "></i>
            <span>الأساسيات</span>
          </div>
          <div class="sidebar-nav-list" >
            
            
              <div class="sidebar-nav-item">
                1. <a class="sidebar-lesson  active" href="/ar/lessons/basics/basics/">الأساسيات</a>
              </div>
            
          </div>
        </div>
      
        
        
        <div class="sidebar-nav-group advanced">
          <div class="sidebar-nav-header">
            <i class="fa  fa-chevron-right "></i>
            <span>الموضوعات المتقدّمة</span>
          </div>
          <div class="sidebar-nav-list"  style="display: none;" >
            
            
          </div>
        </div>
      
        
        
        <div class="sidebar-nav-group specifics">
          <div class="sidebar-nav-header">
            <i class="fa  fa-chevron-right "></i>
            <span>التفاصيل</span>
          </div>
          <div class="sidebar-nav-list"  style="display: none;" >
            
            
          </div>
        </div>
      
    </nav>

    <p>
      

      
        <a href=" /lessons/basics/basics/ ">en</a>
      
        <a href=" /jp/lessons/basics/basics/ ">jp</a>
      
        <a href=" /cn/lessons/basics/basics/ ">cn</a>
      
        <a href=" /es/lessons/basics/basics/ ">es</a>
      
        <a href=" /pt/lessons/basics/basics/ ">pt</a>
      
        <a href=" /vi/lessons/basics/basics/ ">vi</a>
      
        <a href=" /ru/lessons/basics/basics/ ">ru</a>
      
        <a href=" /sk/lessons/basics/basics/ ">sk</a>
      
        <a href=" /fr/lessons/basics/basics/ ">fr</a>
      
        <a href=" /gr/lessons/basics/basics/ ">gr</a>
      
        <a href=" /it/lessons/basics/basics/ ">it</a>
      
        <a href=" /pl/lessons/basics/basics/ ">pl</a>
      
        <a href=" /my/lessons/basics/basics/ ">my</a>
      
        <a href=" /no/lessons/basics/basics/ ">no</a>
      
        <a href=" /ko/lessons/basics/basics/ ">ko</a>
      
        <a href=" /id/lessons/basics/basics/ ">id</a>
      
        <a href=" /ar/lessons/basics/basics/ ">ar</a>
      
    </p>

    <p>&copy; 2016 <a href="https://github.com/doomspork">Sean Callan</a></p>
  </div>
</div>


    <div class="content container">
      <div class="page">
  <h1 class="page-title">الأساسيات</h1>
  <p>البدء، أنواع البيانات الأساسية والعمليات الأساسية.</p>

<h2>جدول المحتويات</h2>
<div id="toc"></div>

<h2 id="section">البداء</h2>

<h3 id="section-1">تثبيت إليكسير</h3>

<p>يمكن إيجاد تعليمات التثبيت لكل نظام التشغيل على موقع elixir-lang.org في دليل <a href="http://elixir-lang.org/install.html">تثبيت إليكسير</a>.</p>

<p>After Elixir is installed you can easily confirm the installed version.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>% elixir -v
Erlang/OTP 18 [erts-7.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.2.3
</code></pre>
</div>

<h3 id="trying-interactive-mode">Trying Interactive Mode</h3>

<p>Elixir comes with <code class="highlighter-rouge">iex</code>, an interactive shell, which allows us to evaluate Elixir expressions as we go.</p>

<p>To get started, let’s run <code class="highlighter-rouge">iex</code>:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>Erlang/OTP 18 [erts-7.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Interactive Elixir (1.2.3) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)&gt;
</code></pre>
</div>

<p>Lets go ahead and give it a try now by typing in a few simple expressions:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>iex(1)&gt; 2+3
5
iex(2)&gt; 2+3 == 5
true
iex(3)&gt; String.length("The quick brown fox jumps over the lazy dog")
43
</code></pre>
</div>

<p>Don’t worry if you don’t understand every expression yet, but hopefully you get the idea.</p>

<h2 id="basic-data-types">Basic Data Types</h2>

<h3 id="integers">Integers</h3>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="m">255</span>
<span class="m">255</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">0xFF</span>
<span class="m">255</span>
</code></pre>
</div>

<p>Support for binary, octal, and hexadecimal numbers comes built in:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="m">0b0110</span>
<span class="m">6</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="err">0</span><span class="n">o644</span>
<span class="m">420</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">0x1F</span>
<span class="m">31</span>
</code></pre>
</div>

<h3 id="floats">Floats</h3>

<p>In Elixir, float numbers require a decimal after at least one digit; they have 64 bit double precision and support <code class="highlighter-rouge">e</code> for exponent numbers:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="m">3.41</span>
<span class="m">3.41</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="o">.</span><span class="m">41</span>
<span class="o">**</span> <span class="p">(</span><span class="no">SyntaxError</span><span class="p">)</span> <span class="ss">iex:</span><span class="m">2</span><span class="p">:</span> <span class="n">syntax</span> <span class="n">error</span> <span class="ss">before:</span> <span class="s1">'.'</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">1.0e-10</span>
<span class="m">1.0e-10</span>
</code></pre>
</div>

<h3 id="booleans">Booleans</h3>

<p>Elixir supports <code class="highlighter-rouge">true</code> and <code class="highlighter-rouge">false</code> as booleans; everything is truthy except for <code class="highlighter-rouge">false</code> and <code class="highlighter-rouge">nil</code>:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="no">true</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="no">false</span>
<span class="no">false</span>
</code></pre>
</div>

<h3 id="atoms">Atoms</h3>

<p>An atom is a constant whose name is their value. If you’re familiar with Ruby these are synonymous with Symbols:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="ss">:foo</span>
<span class="ss">:foo</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="ss">:foo</span> <span class="o">==</span> <span class="ss">:bar</span>
<span class="no">false</span>
</code></pre>
</div>

<p>NOTE: Booleans <code class="highlighter-rouge">true</code> and <code class="highlighter-rouge">false</code> are also the atoms <code class="highlighter-rouge">:true</code> and <code class="highlighter-rouge">:false</code> respectively.</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="no">true</span> <span class="o">|&gt;</span> <span class="n">is_atom</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="ss">:true</span> <span class="o">|&gt;</span> <span class="n">is_boolean</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="ss">:true</span> <span class="o">===</span> <span class="no">true</span>
<span class="no">true</span>
</code></pre>
</div>

<h3 id="strings">Strings</h3>

<p>Strings in Elixir are UTF-8 encoded and are wrapped in double quotes:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="sd">"</span><span class="s2">Hello"</span>
<span class="sd">"</span><span class="s2">Hello"</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="sd">"</span><span class="s2">dziękuję"</span>
<span class="sd">"</span><span class="s2">dziękuję"</span>
</code></pre>
</div>

<p>Strings support line breaks and escape sequences:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="sd">"</span><span class="s2">foo
...&gt; bar"</span>
<span class="sd">"</span><span class="s2">foo\nbar"</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="sd">"</span><span class="s2">foo\nbar"</span>
<span class="sd">"</span><span class="s2">foo\nbar"</span>
</code></pre>
</div>

<p>Elixir also includes more complex data types.  We’ll learn more about these when we learn about Collections and Functions.</p>

<h2 id="basic-operations">Basic Operations</h2>

<h3 id="arithmetic">Arithmetic</h3>

<p>Elixir supports the basic operators <code class="highlighter-rouge">+</code>, <code class="highlighter-rouge">-</code>, <code class="highlighter-rouge">*</code>, and <code class="highlighter-rouge">/</code> as you would expect.  It’s important to notice that <code class="highlighter-rouge">/</code> will always return a float:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="m">2</span> <span class="o">+</span> <span class="m">2</span>
<span class="m">4</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">2</span> <span class="o">-</span> <span class="m">1</span>
<span class="m">1</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">2</span> <span class="o">*</span> <span class="m">5</span>
<span class="m">10</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">10</span> <span class="o">/</span> <span class="m">5</span>
<span class="m">2.0</span>
</code></pre>
</div>

<p>If you need integer division or the division remainder, Elixir comes with two helpful functions to achieve this:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="n">div</span><span class="p">(</span><span class="m">10</span><span class="p">,</span> <span class="m">5</span><span class="p">)</span>
<span class="m">2</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="n">rem</span><span class="p">(</span><span class="m">10</span><span class="p">,</span> <span class="m">3</span><span class="p">)</span>
<span class="m">1</span>
</code></pre>
</div>

<h3 id="boolean">Boolean</h3>

<p>Elixir provides the <code class="highlighter-rouge">||</code>, <code class="highlighter-rouge">&amp;&amp;</code>, and <code class="highlighter-rouge">!</code> boolean operators. These support any types:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="o">-</span><span class="m">20</span> <span class="o">||</span> <span class="no">true</span>
<span class="o">-</span><span class="m">20</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="no">false</span> <span class="o">||</span> <span class="m">42</span>
<span class="m">42</span>

<span class="n">iex</span><span class="o">&gt;</span> <span class="m">42</span> <span class="o">&amp;&amp;</span> <span class="no">true</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">42</span> <span class="o">&amp;&amp;</span> <span class="no">nil</span>
<span class="no">nil</span>

<span class="n">iex</span><span class="o">&gt;</span> <span class="n">!42</span>
<span class="no">false</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="n">!false</span>
<span class="no">true</span>
</code></pre>
</div>

<p>There are three additional operators whose first argument <em>must</em> be a boolean (<code class="highlighter-rouge">true</code> and <code class="highlighter-rouge">false</code>):</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="no">true</span> <span class="ow">and</span> <span class="m">42</span>
<span class="m">42</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="no">false</span> <span class="ow">or</span> <span class="no">true</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="ow">not</span> <span class="no">false</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">42</span> <span class="ow">and</span> <span class="no">true</span>
<span class="o">**</span> <span class="p">(</span><span class="no">ArgumentError</span><span class="p">)</span> <span class="n">argument</span> <span class="ss">error:</span> <span class="m">42</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="ow">not</span> <span class="m">42</span>
<span class="o">**</span> <span class="p">(</span><span class="no">ArgumentError</span><span class="p">)</span> <span class="n">argument</span> <span class="n">error</span>
</code></pre>
</div>

<h3 id="comparison">Comparison</h3>

<p>Elixir comes with all the comparisons operators we’re used to: <code class="highlighter-rouge">==</code>, <code class="highlighter-rouge">!=</code>, <code class="highlighter-rouge">===</code>, <code class="highlighter-rouge">!==</code>, <code class="highlighter-rouge">&lt;=</code>, <code class="highlighter-rouge">&gt;=</code>, <code class="highlighter-rouge">&lt;</code> and <code class="highlighter-rouge">&gt;</code>.</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="m">1</span> <span class="o">&gt;</span> <span class="m">2</span>
<span class="no">false</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">1</span> <span class="o">!=</span> <span class="m">2</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">2</span> <span class="o">==</span> <span class="m">2</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">2</span> <span class="o">&lt;=</span> <span class="m">3</span>
<span class="no">true</span>
</code></pre>
</div>

<p>For strict comparison of integers and floats use <code class="highlighter-rouge">===</code>:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="m">2</span> <span class="o">==</span> <span class="m">2.0</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="m">2</span> <span class="o">===</span> <span class="m">2.0</span>
<span class="no">false</span>
</code></pre>
</div>

<p>An important feature of Elixir is that any two types can be compared; this is particularly useful in sorting.  We don’t need to memorize the sort order, but it is important to be aware of it:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">number</span> <span class="o">&lt;</span> <span class="n">atom</span> <span class="o">&lt;</span> <span class="n">reference</span> <span class="o">&lt;</span> <span class="n">functions</span> <span class="o">&lt;</span> <span class="n">port</span> <span class="o">&lt;</span> <span class="n">pid</span> <span class="o">&lt;</span> <span class="n">tuple</span> <span class="o">&lt;</span> <span class="n">maps</span> <span class="o">&lt;</span> <span class="n">list</span> <span class="o">&lt;</span> <span class="n">bitstring</span>
</code></pre>
</div>

<p>This can lead to some interesting, yet valid, comparisons you may not find in other languages:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="ss">:hello</span> <span class="o">&gt;</span> <span class="m">999</span>
<span class="no">true</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="p">{</span><span class="ss">:hello</span><span class="p">,</span> <span class="ss">:world</span><span class="p">}</span> <span class="o">&gt;</span> <span class="p">[</span><span class="m">1</span><span class="p">,</span> <span class="m">2</span><span class="p">,</span> <span class="m">3</span><span class="p">]</span>
<span class="no">false</span>
</code></pre>
</div>

<h3 id="string-interpolation">String Interpolation</h3>

<p>If you’ve used Ruby, string interpolation in Elixir will look familiar:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="n">name</span> <span class="o">=</span> <span class="sd">"</span><span class="s2">Sean"</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="sd">"</span><span class="s2">Hello </span><span class="si">#{</span><span class="n">name</span><span class="si">}</span><span class="s2">"</span>
<span class="sd">"</span><span class="s2">Hello Sean"</span>
</code></pre>
</div>

<h3 id="string-concatenation">String Concatenation</h3>

<p>String concatenation uses the <code class="highlighter-rouge">&lt;&gt;</code> operator:</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="n">iex</span><span class="o">&gt;</span> <span class="n">name</span> <span class="o">=</span> <span class="sd">"</span><span class="s2">Sean"</span>
<span class="n">iex</span><span class="o">&gt;</span> <span class="sd">"</span><span class="s2">Hello "</span> <span class="o">&lt;&gt;</span> <span class="n">name</span>
<span class="sd">"</span><span class="s2">Hello Sean"</span>
</code></pre>
</div>


  <hr />
<h3 id="social">Share This Page</h3>
<div class="social__container">
  <div class="social__item">
    <a target="_blank" href="https://www.linkedin.com/shareArticle?mini=true&url=https://elixirschool.com/ar/lessons/basics/basics/&title=الأساسيات&summary=description&source=https://elixirschool.com" class="social__icon--linkedin" style="color: #fff; text-decoration: none;"><i class="icon--linkedin"></i></a>
  </div>
  <div class="social__item">
    <a target="_blank" href="https://twitter.com/intent/tweet?url=https://elixirschool.com/ar/lessons/basics/basics/&via=elixirschool&text=ElixirSchool: الأساسيات&hashtags=learnelixir%2Celixirlang&" class="social__icon--twitter" style="color: #fff; text-decoration: none;"><i class="icon--twitter"></i></a>
  </div>
  <div class="social__item">
    <a target="_blank" href="https://plus.google.com/share?url=https://elixirschool.com/ar/lessons/basics/basics/" class="social__icon--googleplus" style="color: #fff; text-decoration: none;"><i class="icon--googleplus"></i></a>
  </div>
  <div class="social__item">
    <a target="_blank" href="https://www.facebook.com/sharer/sharer.php?u=https://elixirschool.com/ar/lessons/basics/basics/" class="social__icon--facebook" style="color: #fff; text-decoration: none;"><i class="icon--facebook"></i></a>
  </div>
  <div class="social__item">
    <a target="_blank" href="https://www.pinterest.com/pin/create/link/?url=https://elixirschool.com/ar/lessons/basics/basics/&media=https://elixirschool.com/assets/fb_share.jpg&description=ElixirSchool: الأساسيات" class="social__icon--pintrest" style="color: #fff; text-decoration: none;"><i class="icon--pintrest"></i></a>
  </div>
  <div class="social__item">
    <a target="_blank" href="https://www.evernote.com/noteit.action?url=https://elixirschool.com/ar/lessons/basics/basics/&title=ElixirSchool: الأساسيات&suggestTags=elixir,programming" class="social__icon--evernote" style="color: #fff; text-decoration: none;"><i class="icon--evernote"></i></a>
  </div>
  <div class="social__item">
    <a target="_blank" href="mailto:?&subject=ElixirSchool: الأساسيات&body=Check out 'الأساسيات' on ElixirSchool%0D%0A%0D%0Ahttps://elixirschool.com/ar/lessons/basics/basics/" class="social__icon--email" style="color: #fff; text-decoration: none;"><i class="icon--email"></i></a>
  </div>
  <div class="social__item">
    <a href="javascript:window.print()" class="social__icon--print" style="color: #fff; text-decoration: none;"><i class="icon--print"></i></a>
  </div>
</div>
</div>
    </div>
  <script type="text/javascript" src="/assets/app.js"></script>
  </body>
</html>
```

**Comment by jaksz** [comment]
Thanks for the file. 

I have tried out different things, and this is what seems to work:

In the CSS file we add something like

```
h1, h2, h3, h4, p {
    direction: rtl;
}

code {
    direction: rtl;
    text-alignment: left;
}
```

We have to make sure that the page width is not too large or the document will look weird.

I know that using CSS to set `dir` is not necessary the most [beautiful way](http://www.w3.org/International/questions/qa-bidi-css-markup) but I think it will do its job here.

However, this needs to be set for the Arabic files only, as setting it globally would mess with all the other languages. How can we do this best?

**Comment by nscyclone** [comment]
@jaksz, there are several languages with right-to-left direction and this might become useful for those languages in the future. So I think we could add an option with the list into `config.yml`. 
```yaml
rtl_languages: ['ar']
```
Then we could add the stylesheets listed above:
```css
/**
 * Right-to-left languages
 */
body.rtl, body.rtl code {
    direction: rtl;
    text-alignment: left;
}
```
Then using Liquid we could apply the class when needed:
```html
{% if site.rtl_languages contains page.lang %}
  {% assign body_class = 'rtl' %}
{% endif %}
<body class="{{ body_class }}">
```
This is more schematic than real and I might be missing something.
@doomspork, what do you think about this?

**Comment by doomspork** [comment]
@nscyclone I think that's a suitable suggestion, would you be open to submitting a PR?

**Comment by nscyclone** [comment]
@doomspork, sure, I'll take a closer look at this soon.

**Comment by nscyclone** [comment]
@jaksz, could you check if [this](http://imgur.com/a/3OVmG) is correct once you have a moment?

**Comment by jaksz** [comment]
Hi @nscyclone, thanks for the image.

## ✅ ~~Original problem described in first post~~
Thankfully this is fixed, the text is at least generally readable.

## 🚫  Alignment
Everything should be aligned to the right, at the moment it's aligned to the left. This is actually quite important because, as visible in this example, there is a "hangover" word from the sentence in the new line, but at the end of the new line (for an Arabic speaker) instead of the beginning:
qpvipgb


## ❓  Out-of-place characters
I assume these characters here are leftovers from the Markdown formatting? Or how did they end up here? 
ytoj7vf

**Comment by nscyclone** [comment]
@jaksz, unfortunately, I couldn't find a way to make the hash syntax for headers work. The only workaround I've found for now is plain HTML (note that `id` is required for header in order to be included in automatic TOC generation):
<details> 
```
---
layout: page
title: الأساسيات
category: basics
order: 1
lang: ar
---
+البدء، أنواع البيانات الأساسية والعمليات الأساسية.
{% include toc.html %}
<h2 id="getting-started">البداء</h2>
<h3 id="installing-elixir">تثبيت إليكسير</h3>
يمكن إيجاد تعليمات التثبيت لكل نظام التشغيل على موقع elixir-lang.org في دليل [تثبيت إليكسير](http://elixir-lang.org/install.html).
```
</details>
Here's the [output](http://i.imgur.com/6bdIFwk.png?1). What do you think about this?

**Comment by jaksz** [comment]
@nscyclone, thanks for the output, everything looks fine now. However, we need to check how it looks with the code. Can you render the new commit?

Also, does that mean I cannot write the translation in pure Markdown, but need to manually include <h2>, <h3>, ..., tags? Is there a way to automate this? Does this problem exist with all translations or just Arabic?

**Comment by nscyclone** [comment]
@jaksz, eventually, the atx-style headers started working with the latest file. So I've replaced the tags with hashes back. I yet do not know what caused the problem before so we might need some sort of tests not to have unwanted surprises later. 
Here's the [output](http://i.imgur.com/UZntSZK.png). Code section looks quite strange and it probably needs some sort of correction. Do we just need to align it left?

**Comment by jaksz** [comment]
@nscyclone, thanks for the great news, this makes the translation process more user-friendly.

and yes, making the code align left will fix this issue!

**Comment by nscyclone** [comment]
@jaksz, I've just fixed the code alignment. Now the whole page looks like [this](http://imgur.com/a/Avgfp).  Is there anything else to be improved? If this is sufficient I am ready to propose a PR.

**Comment by jaksz** [comment]
@nscyclone this looks perfect. Can you show all the code that needs to be included in the `.md` file?

I suggest I finish translating the `basics.md` file and then we can merge it.

**Comment by nscyclone** [comment]
@jaksz, I've just sent a PR: #442. And I guess there is nothing to be included in the actual translations. But we still need the changes you made in `_config.yml` to add the Arabic language because I just included it into the new `rtl_languages` option and did nothing else.

**Comment by jaksz** [comment]
@nscyclone great, and I have just completed translating `basics.md`. Looking forward to seeing it in action to check for any remaining issues.

**Comment by doomspork** [comment]
@jaksz I've merged in @nscyclone's changes, you should be good to rebase 👍

**Comment by doomspork** [comment]
Hi @jaksz, are you still interested in getting these changes merged in?  We need to rebase your branch to merge.

**Comment by doomspork** [comment]
@jaksz could you rebase your branch or merge master in so we could merge?  I'd love to get this live.

**Comment by doomspork** [comment]
Happy New Year @jaksz!  I hope you're doing well.  Do you wish to continue your translation work?  We've updated the code to support the necessary changes 😀

**Comment by jaksz** [comment]
Happy New Year to you, too, @doomspork !

Did that do what you wanted? Is that enough to put the first article live?

If everything works so far from the technical side, I'll do my best to continue the translation ;)

**Comment by brain-geek** [comment]
@jaksz @doomspork I can rebase this one and merge, if you don't mind.

I think that having one more language is beneficial even with one lesson.

**Comment by doomspork** [comment]
@brain-geek that works for me if @jaksz is okay with that 👍

**Comment by doomspork** [comment]
@brain-geek did you want to take this one and rebase or should I take care of that?

**Comment by nscyclone** [comment]
@doomspork I'm afraid @brain-geek does not seem to be active on GitHub lately. So I decided to open [a new PR](https://github.com/elixirschool/elixirschool/pull/1174), because some additional changes were required for this to work.

**Comment by brain-geek** [comment]
Yep, you're right @nscyclone . Thanks for doing this! I'll close this PR.

**Comment by doomspork** [comment]
Thank you both 👍

**Pull request closed without merging**